### PR TITLE
Early Polling Locations

### PR DESCRIPTION
**Pull request by lucyq**
Added markers to display early polling locations
Info windows show dates/times that locations will be open
Additional styling is needed

### CODE REVIEW

File: app/scripts/polling_location_finder.js [hunk 1]
```diff
@@ -78,8 +86,101 @@ define(['jquery', 'geojson', 'json!vendor/ELECTIONS_WardsPrecincts.geojson', 'js
...
+            map: map
+        });
+
+        var earlyVotingInfoWindow = new google.maps.InfoWindow();
+
+        var earlyVotingInfo = document.createElement("ul");
+        earlyVotingInfo.id = "earlyVotingInfoDiv";
+
+        var hoursHTML = generateHoursHTML(listed_hours);
+        earlyVotingInfo = "<div class='earlyPollingWindow'><b><h5 style='color:#000'>Early Polling Location: "
```
Comment (carpeliam) [inline app/scripts/polling_location_finder.js]: this removes the list you created on line 159 - not sure if this is what you meant to do...

File: app/scripts/polling_location_finder.js [hunk 2]
```diff
@@ -78,8 +86,101 @@ define(['jquery', 'geojson', 'json!vendor/ELECTIONS_WardsPrecincts.geojson', 'js
...
+                earlyVotingInfoWindow.open(map, this);
+            }
+        })(earlyVotingInfo, index));    
+
+        google.maps.event.addListener(map, 'click', function() {
+            earlyVotingInfoWindow.close();
+        });
+    }
+
+    return function(latLng) {
```
Comment (carpeliam) [inline app/scripts/polling_location_finder.js]: The previous return function took a success and error callback - I think we still need those, unless I'm missing some magic :)

**Comment by carpeliam** [comment]
(mentioning for tracking purposes that this PR is for #31)

File: app/scripts/polling_location_finder.js [hunk 3]
```diff
@@ -78,8 +86,101 @@ define(['jquery', 'geojson', 'json!vendor/ELECTIONS_WardsPrecincts.geojson', 'js
...
     }
 
-    return function(latLng, successCallback, errorCallback) {
+    function getHours(place) {
+
+        var hours = [];
+        moment.tz.add('America/New_York|EST EDT|50 40|0101|1Lz50 1zb0 Op0');
+        
+        jQuery.each(place, function(index, timeInterval) {
+            var range = moment.range(timeInterval);
```
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: Hmmm I think you're right about this! It wasn't originally working but that might've been for other reasons

File: app/scripts/polling_location_finder.js (hunk 1 above)
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: yup! I was originally going to style it differently and forgot to delete that line

File: app/scripts/polling_location_finder.js (hunk 2 above)
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: which function?

File: bower.json [hunk 4]
```diff
@@ -7,7 +7,9 @@
     "jquery": "~1.10.2",
     "requirejs-text": "~2.0.10",
     "geojson-google-maps": "*",
-    "font-awesome": "~4.0.2"
+    "font-awesome": "~4.0.2",
+    "moment": "^2.15.1",
+    "moment-timezone": "^0.5.5"
```
Comment (lucyq) [inline bower.json]: yes, it needs to be included!

File: app/scripts/polling_location_finder.js [hunk 5]
```diff
@@ -78,8 +86,101 @@ define(['jquery', 'geojson', 'json!vendor/ELECTIONS_WardsPrecincts.geojson', 'js
         return encodeURI(url + destination);
     }
 
-    return function(latLng, successCallback, errorCallback) {
+    function getHours(place) {
+
+        var hours = [];
+        moment.tz.add('America/New_York|EST EDT|50 40|0101|1Lz50 1zb0 Op0');
+        
+        jQuery.each(place, function(index, timeInterval) {
```
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: yeah! i'll change this

File: app/scripts/polling_location_finder.js (hunk 3 above)
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: nevermind. just tried removing them and the functions stopped working.

Comment (carpeliam) [inline app/scripts/polling_location_finder.js]: Haha oh well! This is super unimportant, but I'm curious to toy around with it at some point... or just move away from RequireJS altogether...

File: app/scripts/polling_location_finder.js (hunk 2 above)
Comment (carpeliam) [inline app/scripts/polling_location_finder.js]: The function that this file returns used to look like `return function(latLng, successCallback, errorCallback)`. We're still passing callbacks in, but it doesn't look like we're using them now, unless I'm missing something.

**Comment by thecristen** [comment]
Sorry for asking this so late, but might you be able to add screenshots? Node's not really working on my work computer and I'd love to see more!

File: app/scripts/polling_location_finder.js (hunk 2 above)
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: hmmm i don't remember ever taking those out but can add that back in!

Comment (lucyq) [inline app/scripts/polling_location_finder.js]: i'm not seeing where these callbacks are defined though :/

**Comment by lucyq** [comment]
Refactored code for parsing early polling hours! Check out new commit

File: app/scripts/polling_location_finder.js (hunk 2 above)
Comment (lucyq) [inline app/scripts/polling_location_finder.js]: nevermind! Didn't have the updated file but now see the line you're talking about & that they're getting passed in from app.js!

**Comment by lucyq** [comment]
Ok, just pushed again! moment-timezone and moment-range are defined but no longer formal parameters. callbacks are back!

**Comment by carpeliam** [comment]
Hey @lucyq! I'm thinking this PR might not make sense in its current form now that we've formed a direction with #50. I'm thinking about closing this PR, but hopefully some of the code can be used toward a new PR against the current master. Thoughts?

**Comment by lucyq** [comment]
yup, makes sense! i'll start working on getting parts of it integrated w/ the new changes!

**Comment by carpeliam** [comment]
It's actually kind of awesome that we have 3 PRs all for early voting. I'll work on pulling work from each of them in, and then close them each afterward. Stay tuned!

**Comment by carpeliam** [comment]
Now that we have early voting locations in master, I'm going to close this PR. We can still use some code from it as necessary within the scope of a new PR, but I think it's safe to close this one. If this is premature, please add some further comments below.

**Pull request closed without merging**